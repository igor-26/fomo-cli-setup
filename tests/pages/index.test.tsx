import Index from 'pages/index'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'

jest.mock('next/router', () => ({
	useRouter() {
		return {
			asPath: '/',
		}
	},
}))

describe('Index page', () => {
	it('renders without crashing', () => {
		render(<Index />)
	})

	it('has a heading, GitHub star and a link to /integrations', () => {
		render(<Index />)

		expect(
			screen.getByRole('heading', {
				name: /Browse social media from the comfort of your terminal./i,
			})
		)
		expect(screen.getByLabelText('Star igor-26/fomo-cli on GitHub'))
		expect(
			screen.getByRole('link', { name: /set up integrations/i })
		).toHaveAttribute('href', '/integrations')
	})

	it('matches snapshot', () => {
		const { container } = render(<Index />)
		expect(container).toMatchSnapshot()
	})
})
