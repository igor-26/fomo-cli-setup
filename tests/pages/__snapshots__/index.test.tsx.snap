// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col justify-between min-h-screen"
  >
    <div
      class="border-b border-gray-200 dark:border-gray-800 pb-8 relative flex-1"
    >
      <div
        class="container mx-auto p-4"
      >
        <header>
          <nav
            class="mb-8 flex justify-center items-center"
          >
            <ul
              class="flex gap-4 border border-gray-200 dark:border-gray-600 items-center rounded-full pl-4 pr-1 py-1"
            >
              <li>
                <a
                  class="hover:text-accent-400 text-accent-400"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  class="hover:text-accent-400 "
                  href="/integrations"
                >
                  Setup
                </a>
              </li>
              <li>
                <a
                  class="hover:text-accent-400 inline-flex"
                  href="https://github.com/igor-26/fomo-cli"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="inline-flex items-center"
                  >
                    <span
                      class="mr-0.5"
                    >
                      GitHub
                    </span>
                    <svg
                      aria-label="External link icon"
                      class="w-4 h-4"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M4.25 5.5a.75.75 0 00-.75.75v8.5c0 .414.336.75.75.75h8.5a.75.75 0 00.75-.75v-4a.75.75 0 011.5 0v4A2.25 2.25 0 0112.75 17h-8.5A2.25 2.25 0 012 14.75v-8.5A2.25 2.25 0 014.25 4h5a.75.75 0 010 1.5h-5z"
                        fill-rule="evenodd"
                      />
                      <path
                        clip-rule="evenodd"
                        d="M6.194 12.753a.75.75 0 001.06.053L16.5 4.44v2.81a.75.75 0 001.5 0v-4.5a.75.75 0 00-.75-.75h-4.5a.75.75 0 000 1.5h2.553l-9.056 8.194a.75.75 0 00-.053 1.06z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </a>
              </li>
              <li
                class="hover:opacity-80 flex bg-gray-50 dark:bg-gray-800 rounded-full border border-gray-300 dark:border-gray-600 p-0.5"
              >
                <button
                  title="theme-mode"
                >
                  <svg
                    aria-hidden="true"
                    class="w-5 h-5 rounded-full"
                    fill="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 2.25a.75.75 0 01.75.75v2.25a.75.75 0 01-1.5 0V3a.75.75 0 01.75-.75zM7.5 12a4.5 4.5 0 119 0 4.5 4.5 0 01-9 0zM18.894 6.166a.75.75 0 00-1.06-1.06l-1.591 1.59a.75.75 0 101.06 1.061l1.591-1.59zM21.75 12a.75.75 0 01-.75.75h-2.25a.75.75 0 010-1.5H21a.75.75 0 01.75.75zM17.834 18.894a.75.75 0 001.06-1.06l-1.59-1.591a.75.75 0 10-1.061 1.06l1.59 1.591zM12 18a.75.75 0 01.75.75V21a.75.75 0 01-1.5 0v-2.25A.75.75 0 0112 18zM7.758 17.303a.75.75 0 00-1.061-1.06l-1.591 1.59a.75.75 0 001.06 1.061l1.591-1.59zM6 12a.75.75 0 01-.75.75H3a.75.75 0 010-1.5h2.25A.75.75 0 016 12zM6.697 7.757a.75.75 0 001.06-1.06l-1.59-1.591a.75.75 0 00-1.061 1.06l1.59 1.591z"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </nav>
        </header>
        <main>
          <main
            class="pt-16 lg:flex text-center lg:text-left"
          >
            <div
              class="lg:w-2/3"
            >
              <div
                class="mb-12 lg:max-w-xl lg:pr-16"
              >
                <h1
                  class="font-bold mb-8 text-5xl tracking-tighter"
                >
                  Browse social media from the comfort of your
                   
                  <span
                    class="text-accent-400"
                  >
                    terminal
                  </span>
                  .
                </h1>
                <p
                  class="text-zinc-500 text-lg"
                >
                  No ads, no suggested content, no fluff.
                  <br />
                  Only the content you are subscribed to.
                </p>
              </div>
              <div
                class="mb-16 flex justify-center lg:justify-start items-center"
              >
                <a
                  class="font-semibold bg-accent-400 hover:opacity-90 rounded-full text-white py-2 font-semi px-5 mr-5 inline-flex"
                  href="/integrations"
                >
                  Set up integrations
                </a>
                <div
                  class="pt-2"
                >
                  <span>
                    <a
                      aria-label="Star igor-26/fomo-cli on GitHub"
                      data-icon="octicon-star"
                      data-show-count="true"
                      data-size="large"
                      href="https://github.com/igor-26/fomo-cli"
                    >
                      Star
                    </a>
                    <span />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="border border-gray-200 flex rounded-md lg:rounded-xl mb-3"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27900%27%20height=%27529%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="fomo"
                  class="rounded-md lg:rounded-xl"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="/_next/image?url=%2Fimages%2Ffomo.gif&w=1920&q=75"
                  srcset="/_next/image?url=%2Fimages%2Ffomo.gif&w=1080&q=75 1x, /_next/image?url=%2Fimages%2Ffomo.gif&w=1920&q=75 2x"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </span>
            </div>
          </main>
        </main>
      </div>
    </div>
    <div
      class="container mx-auto"
    >
      <footer
        class="pt-8 pb-16 px-4 md:flex text-center justify-between items-center"
      >
        <div
          class="flex justify-center md:justify-left mb-3 md:mb-0"
        >
          <ul
            class="flex gap-3"
          >
            <li>
              <a
                class="hover:text-accent-400"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class="hover:text-accent-400"
                href="/integrations"
              >
                Setup
              </a>
            </li>
            <li>
              <a
                class="hover:text-accent-400 inline-flex"
                href="https://github.com/igor-26/fomo-cli"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="inline-flex items-center"
                >
                  <span
                    class="mr-0.5"
                  >
                    GitHub
                  </span>
                  <svg
                    aria-label="External link icon"
                    class="w-4 h-4"
                    fill="currentColor"
                    viewBox="0 0 20 20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M4.25 5.5a.75.75 0 00-.75.75v8.5c0 .414.336.75.75.75h8.5a.75.75 0 00.75-.75v-4a.75.75 0 011.5 0v4A2.25 2.25 0 0112.75 17h-8.5A2.25 2.25 0 012 14.75v-8.5A2.25 2.25 0 014.25 4h5a.75.75 0 010 1.5h-5z"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M6.194 12.753a.75.75 0 001.06.053L16.5 4.44v2.81a.75.75 0 001.5 0v-4.5a.75.75 0 00-.75-.75h-4.5a.75.75 0 000 1.5h2.553l-9.056 8.194a.75.75 0 00-.053 1.06z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </a>
            </li>
          </ul>
        </div>
        <p
          class="text-sm text-zinc-400 dark:text-zinc-500 copyright-container"
        >
          Â© 2022
           
          <a
            class="hover:text-accent-400 inline-flex"
            href="https://github.com/igor-26"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="inline-flex items-center"
            >
              <span
                class="mr-0.5"
              >
                igor-26
              </span>
              <svg
                aria-label="External link icon"
                class="w-4 h-4"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4.25 5.5a.75.75 0 00-.75.75v8.5c0 .414.336.75.75.75h8.5a.75.75 0 00.75-.75v-4a.75.75 0 011.5 0v4A2.25 2.25 0 0112.75 17h-8.5A2.25 2.25 0 012 14.75v-8.5A2.25 2.25 0 014.25 4h5a.75.75 0 010 1.5h-5z"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M6.194 12.753a.75.75 0 001.06.053L16.5 4.44v2.81a.75.75 0 001.5 0v-4.5a.75.75 0 00-.75-.75h-4.5a.75.75 0 000 1.5h2.553l-9.056 8.194a.75.75 0 00-.053 1.06z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </a>
           
          Â· Built with
           
          <a
            class="hover:text-accent-400 inline-flex"
            href="https://nextjs.org/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="inline-flex items-center"
            >
              <span
                class="mr-0.5"
              >
                Next.js
              </span>
              <svg
                aria-label="External link icon"
                class="w-4 h-4"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4.25 5.5a.75.75 0 00-.75.75v8.5c0 .414.336.75.75.75h8.5a.75.75 0 00.75-.75v-4a.75.75 0 011.5 0v4A2.25 2.25 0 0112.75 17h-8.5A2.25 2.25 0 012 14.75v-8.5A2.25 2.25 0 014.25 4h5a.75.75 0 010 1.5h-5z"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M6.194 12.753a.75.75 0 001.06.053L16.5 4.44v2.81a.75.75 0 001.5 0v-4.5a.75.75 0 00-.75-.75h-4.5a.75.75 0 000 1.5h2.553l-9.056 8.194a.75.75 0 00-.053 1.06z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </a>
        </p>
      </footer>
    </div>
  </div>
</div>
`;
